errbot/plugin_manager.py

```diff
@@ -86,6 +89,95 @@ def check_dependencies(path):
 
 
+def check_python_plug_section(name: str, config: ConfigParser) -> bool:
+    """ Checks if we have the correct version to run this plugin.
+    Returns true if the plugin is loadable """
+    try:
+        python_version = config.get("Python", "Version")
+    except NoSectionError:
+        log.warning(
+            'Plugin %s has no section [Python]. Assuming this '
+            'plugin is runnning only under python 2.', name)
+        python_version = '2'
+
+    if python_version not in ('2', '2+', '3'):
+        log.warning(
+            'Plugin %s has an invalid Version specified in section [Python]. '
+            'The Version can only be 2, 2+ and 3', name)
+        return False
+
+    if python_version == '2' and PY3:
+        log.error(
+            '\nPlugin %s is made for python 2 only and you are running '
+            'err under python 3.\n\n'
+            'If the plugin can be run on python 2 and 3 please add this '
+            'section to its .plug descriptor :\n[Python]\nVersion=2+\n\n'
+            'Or if the plugin is Python 3 only:\n[Python]\nVersion=3\n\n', name)
+        return False
+
+    if python_version == '3' and PY2:
+        log.error('\nPlugin %s is made for python 3 and you are running err under python 2.')
+        return False
+    return True
+
+
+def check_errbot_version(name: str, min_version: str, max_version: str):
+    """ Checks if a plugin version between min_version and max_version is ok
+    for this errbot.
+    Raises IncompatiblePluginException if not.
+    """
+    log.info('Activating %s with min_err_version = %s and max_version = %s',
+             name, min_version, max_version)
+    current_version = version2array(VERSION)
+    if min_version and version2array(min_version) > current_version:
+        raise IncompatiblePluginException(
+            'The plugin %s asks for err with a minimal version of %s while err is version %s' % (
+                name, min_version, VERSION)
+        )
+
+    if max_version and version2array(max_version) < current_version:
+        raise IncompatiblePluginException(
+            'The plugin %s asks for err with a maximal version of %s while err is version %s' % (
+                name, max_version, VERSION)
+        )
+
+
+def check_errbot_plug_section(name: str, config: ConfigParser) -> bool:
+    """ Checks if we have the correct Errbot version.
+    Returns true if the plugin is loadable """
+
+    # Errbot version check
+    try:
+
+        try:
+            min_version = config.get("Errbot", "Min")
+        except NoOptionError:
+            log.debug('Plugin %s has no Min Option in [Errbot] section. '
+                      'Assuming this plugin is running on this Errbot'
+                      'version as min version.', name)
+            min_version = VERSION
+
+        try:
+            max_version = config.get("Errbot", "Max")
+        except NoOptionError:
+            log.debug('Plugin %s has no Min Option in [Errbot] section. '
```

Comment:
Thx @zoni ! fixed in 5d2c8527bc3ea6ac5b213223c7bf60caf149c571 directly on master